### PR TITLE
refac: Use version range for delta-spark instead of hard coded 3.2.0 in setup.py

### PR DESCRIPTION
**Pull request by JesperGomes**

### CODE REVIEW

File: source/spark_sql_migrations/setup.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
     packages=find_packages(),
     install_requires=[
         "pyspark==3.5.1",
-        "delta-spark==3.1.0",
+        "delta-spark>=3.1,<4.0",
```
Comment (BjarkeMeier) [inline source/spark_sql_migrations/setup.py:14]: Nice 👍

File: .docker/requirements.txt [hunk 2]
```diff
@@ -18,4 +18,4 @@ pytest-xdist
 #
 
 dependency_injector == 4.41.0
-delta-spark==3.1.0
+delta-spark>=3.1,<4.0
```
Comment (BjarkeMeier) [inline .docker/requirements.txt:21]: I think the requirement is better off with a specific version. I believe you can revert this change.

Comment (JesperGomes) [reply .docker/requirements.txt]: done

**Review by BjarkeMeier — APPROVED** [review]
